gc/stats/LargeObjectAllocateStats.cpp

```diff
@@ -262,6 +262,33 @@ MM_FreeEntrySizeClassStats::getFreeMemory(const uintptr_t sizeClassSizes[])
 	return totalFreeMemory;
 }
 
+uintptr_t 
+MM_FreeEntrySizeClassStats::getPageAllignedFreeMemory(const uintptr_t sizeClassSizes[], uintptr_t pageSize) {
+
+	uintptr_t resusableFreeMemory = 0;
+
+	for (uintptr_t sizeClassIndex = 0; sizeClassIndex < _maxSizeClasses; sizeClassIndex++) {
+		
+		/* Check if the current size class is larger than the page size */
+		if (pageSize < sizeClassSizes[sizeClassIndex]) {
+
+			/* regular sizes */
+			resusableFreeMemory += _count[sizeClassIndex] * (sizeClassSizes[sizeClassIndex] - pageSize);
+			
+			/* for each size class, find frequent allocation sizes */
+			if (NULL != _frequentAllocationHead) {
+				MM_FreeEntrySizeClassStats::FrequentAllocation *curr = _frequentAllocationHead[sizeClassIndex];
+				while (NULL != curr) {
```

Comment:
Insert a comment here some thing like: 
We need to find how many pages (of certain size, which are aligned to its size) would fit into free entries. We don't have the exact info about start and end of each free entry but only the low bound size and count of such entries. Hence we use a simple heuristic that freeEntrySize - pageSize of bytes, in average will fit. For example if free entry is 6K and page is 4K, in average 2K will be reusable (50% chances it will be 1 full page, and 50% no page will fit).